src/common/transformations/tests/utils/compress_quantize_weights.cpp

```diff
@@ -159,6 +162,41 @@ TEST_F(TransformationTestsF, CompressQuantizeWeightsWithDequantizationSubgraph)
     comparator.enable(FunctionsComparator::CmpValues::ACCURACY);
 }
 
+TEST_F(TransformationTestsF, CompressQuantizeWeightsWithDequantizationSubgraphFP16) {
```

Comment:
Will this test fail without the changes? I don't understand how this mark affects this transformation.

Reply:
yes, it will fail
we use "get_constant_from_source" in the transformation code, and this method checks
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/3b4fdcae-6ade-41fd-9b5a-4655e3ab2a10)
the highlighted method calls "evaluate_both_bounds" method which calls "could_propagate" method:
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/59757c19-5bc8-427c-9eb5-8dfa34477392)
and "could_propagate" checks that 
![image](https://github.com/openvinotoolkit/openvino/assets/21181928/a3e6f0db-f6de-489d-a464-37ef001b8adc)
it's not that obvious at all. 
@pavel-esir could you comment why do we need "is_decompession" check inside the methods above?

Reply:
I added `is_decompression` check inside `ov::could_propagate` in order to disable constant folding for `Const+Convert (with rt_info decompression)` pattern and to do CF on plugin side. I placed this stopper here instead of placing it inside of `ConstantFolding` because some transformations call `get_constant_from_source` and this in the end called expensive `Convert::evaluate`.

At first glance this solution looked ok: this solution removes marking with this attribute calls evaluate and then returns back attribute. But considering that there are question i will double check locally how this solution works

Reply:
It seems quite odd that we have different mechanisms for disabling const_folding:
"disable_const_folding" attribute and this "decompression" attribute (which will be transformed to "disable_const_folding" attribute" via `KeepConstAndDecompression`)

Reply:
I believe this is ok as a quick fix, but we need to discuss it to make the mechanism more clear.
It is necessary to get rid of ambiguity in the functionality of attributes and clearly define their purpose.

Reply:
i tried with `disable_const_folding` but it didn't work for all cases. When `ov::get_constant_from_source()` is called even if `disable_const_folding` is set it didn't prevent from calling `Convert::evaluate()` therefore i added this skipping by `decompression`.

Reply:
Can we have other transformations that need to use `get_constant_from_source `, but wouldn't be able to?

Reply:
ticket: CVS-117367